contracts/hub/router/src/ibc/receive.rs

```diff
@@ -222,22 +227,53 @@ fn execute_request_pool_creation(
         .add_attribute("tx_id", tx_id)
         .add_attribute("method", "request_pool_creation");
 
-    for token in pair.get_vec_token() {
-        let token_exists =
-            ESCROW_BALANCES.has(deps.storage, (token.clone(), sender.clone().chain_uid));
-        let range =
-            ESCROW_BALANCES
-                .prefix(token)
-                .keys_raw(deps.storage, None, None, Order::Ascending);
+    let mut one_token_already_exists = false;
+
+    for token in pair_with_denom.get_vec_token_info() {
+        // Check if token is already validated. Its validated if it has an escrow on sender chain
+        let mut validated_token = ESCROW_BALANCES.has(
+            deps.storage,
+            (token.token.clone(), sender.clone().chain_uid),
+        );
+
+        // Check if token is already present on any chain
+        let range = ESCROW_BALANCES.prefix(token.token).keys_raw(
+            deps.storage,
+            None,
+            None,
+            Order::Ascending,
+        );
+
+        let token_exists_on_any_chain = range.take(1).count() > 0;
+        if token_exists_on_any_chain {
+            one_token_already_exists = true;
+        }
+
+        if token.token_type.is_voucher() {
+            ensure!(
+                token_exists_on_any_chain,
+                ContractError::new(
+                    "Cannot create pool with voucher token that doesn't exist on any chain"
+                )
+            );
+            // Voucher token is valid if it exists on any chain
+            validated_token = true;
+        }
+
         // There are two cases
         // token already exists on the sender chain - We can safely assume that this was validated already by factory so allow pool creation
         // token not present in sender chain -  This token should not have escrow on any other chain, i.e. This should be completely new token
         ensure!(
-            token_exists || range.take(1).count() == 0,
+            validated_token || !token_exists_on_any_chain,
             ContractError::new("Cannot use already existing token without registering it first")
         )
     }
 
+    ensure!(
+        one_token_already_exists,
+        ContractError::new("Cannot create pool with two new tokens")
+    );
+
     // If vlp is already there, send execute msg to it to register the pool, else create a new pool with register msg attached to instantiate msg
```

Comment:
nitkpick: I'd say it would be clearer if we add this line here directly before using it
```rust
let vlp = VLPS.may_load(deps.storage, pair.get_tupple())?;
```